applications/StructuralMechanicsApplication/custom_processes/postprocess_eigenvalues_process.cpp

```diff
@@ -0,0 +1,177 @@
+    }
+
+    void PostprocessEigenvaluesProcess::ExecuteBeforeSolutionLoop()
+    {
+        KRATOS_ERROR_IF_NOT(mpGidEigenIO) << " EigenIO is uninitialized!" << std::endl;
+
+        mpGidEigenIO->InitializeMesh(0.0);
+        mpGidEigenIO->WriteMesh(mrModelPart.GetMesh());
+        mpGidEigenIO->WriteNodeMesh(mrModelPart.GetMesh());
+        mpGidEigenIO->FinalizeMesh();
+    }
+
+    void PostprocessEigenvaluesProcess::ExecuteFinalize()
+    {
+        KRATOS_ERROR_IF_NOT(mpGidEigenIO) << " EigenIO is uninitialized!" << std::endl;
+
+        const auto& eigenvalue_vector = mrModelPart.GetProcessInfo()[EIGENVALUE_VECTOR];
+        const SizeType num_eigenvalues = eigenvalue_vector.size();
+
+        const SizeType num_animation_steps = mOutputParameters["animation_steps"].GetInt();
+        double angle = 0.0;
+        std::string label = "";
+
+        const SizeType num_requested_results = mOutputParameters["eigen_results"].size();
+        std::vector<std::string> requested_results(num_requested_results);
+
+        for (SizeType i=0; i<num_requested_results; ++i)
+            requested_results[i] = mOutputParameters["eigen_results"].GetArrayItem(i).GetString();
+
+        for (SizeType i=0; i < num_animation_steps; ++i)
+        {
+            angle = 2 * Globals::Pi * i / num_animation_steps;
+
+            for(SizeType j=0; j<num_eigenvalues; ++j)
+            {
+                label = GetLabel(eigenvalue_vector[j]);
+
+                for(auto& node : mrModelPart.Nodes())
+                {
+                    // Copy the eigenvector to the Solutionstepvariable. Credit to Michael Andre
+                    DofsContainerType& r_node_dofs = node.GetDofs();
+                    Matrix& r_node_eigenvectors = node.GetValue(EIGENVECTOR_MATRIX);
+
+                    KRATOS_ERROR_IF_NOT(r_node_dofs.size() == r_node_eigenvectors.size2()) 
+                        << "Number of results on node " << node.Id() << " is wrong" << std::endl;
+
+                    SizeType k = 0;
+                    for (auto& r_dof : r_node_dofs)
+                        r_dof.GetSolutionStepValue(0) = std::cos(angle) * r_node_eigenvectors(j,k++);
+                }
+                
+                for (const auto& var_name : requested_results)
+                    WrapperForIOCall(var_name, label, i);                
+            }
+        }
+    }
+
+    std::string PostprocessEigenvaluesProcess::GetLabel(double LabelNumber)
+    {
+        std::string label = "EigenValue_";
+
+        // Compute the Eigenfrequency from the Eigenvalue (if the User specified it)
+        if (mOutputParameters["use_eigenfrequency_in_label"].GetBool())
+        {
+            label = "EigenFrequency_";
+            LabelNumber = std::sqrt(LabelNumber) / (2 * Globals::Pi);
+        }
+
+        std::stringstream strstr;
+        strstr << std::fixed << std::setprecision(3) << LabelNumber;
```

Comment:
Does this work if my eigenvalues are in the range 0.000001,0.000002,... or 10000000?

Reply:
I don't think so
Will change it such that it is more general